### ISSUE TITLE
Constructing a new image w/o a data disk set is a pain...

### ISSUE THREAD

**Issue by DinoV** [Azure/azure-powershell#648]
When constructing a new VM image the code seems to be doing:

```
   var osDiskConfig = DiskConfig == null ? null : DiskConfig.OSDiskConfiguration; 
94                 var dataDiskConfigs = DiskConfig == null ? null : DiskConfig.DataDiskConfigurations.ToList(); 
```
If you have no disk config setup though you end up getting a nasty error:

```
Add-AzureVMImage : Value cannot be null.
Parameter name: source
At line:1 char:1
+ Add-AzureVMImage -OS "Linux" -ImageName $imageName -Label $imageName -DiskConfig ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Add-AzureVMImage], ArgumentNullException
    + FullyQualifiedErrorId : System.ArgumentNullException,Microsoft.WindowsAzure.Commands.ServiceManagement.IaaS.Disk
   Repository.AddAzureVMImage
```

With no indication of what's going on...  The only way I've found of setting the data disk w/o actually constructing a data disk is to do:

$configSet.DataDiskConfigurations = new-object Microsoft.WindowsAzure.Commands.ServiceManagement.Model.DataDiskConfigurationList

Instead of calling DiskConfig.DataDiskConfigurations.ToList() this should do:

(DiskConfig == null || DiskConfig.DataDiskConfigurations == null) ? null : DiskConfig.DataDiskConfigurations.ToList();

**Comment by rhencke** [comment]
Yeah.  We hit the same.  We ended up doing what you're doing.  It's just clumsy to work with.

**Comment by singhkay** [comment]
Fixed in release 1.0.2 https://github.com/Azure/azure-powershell/releases/tag/v1.0.2-December2015
